lib/web_ui/lib/src/engine/semantics/dialog.dart

```diff
@@ -0,0 +1,38 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import '../dom.dart';
+import '../semantics.dart';
+import '../util.dart';
+
+/// Provides accessibility for dialogs.
+///
+/// See also [Role.dialog].
+class Dialog extends RoleManager {
+  Dialog(SemanticsObject semanticsObject) : super(Role.dialog, semanticsObject);
+
+  @override
+  void dispose() {
+    semanticsObject.element.removeAttribute('aria-label');
+    semanticsObject.clearAriaRole();
+  }
+
+  @override
+  void update() {
+    assert(() {
+      final String? label = semanticsObject.label;
+      if (label == null || label.trim().isEmpty) {
+        printWarning(
+          'Semantic node ${semanticsObject.id} was assigned dialog role, but '
+          'is missing a label. A dialog should contain a label so that a '
+          'screen reader can communicate to the user that a dialog appeared '
+          'and a user action is requested.'
+        );
+      }
+      return true;
+    }());
+    semanticsObject.element.setAttribute('aria-label', semanticsObject.label ?? '');
```

Comment:
Do you think we should we have a more descriptive fallback `aria-label` other than an empty string if no label is provided? Or do we generally defer to the dev via warnings like on line 26?

Reply:
We can't because labels are part of the UI, which needs to be internationalized. Error messages and warnings in Flutter, OTOH, are all in English. So I think a warning and missing label is probably a better fallback.